### PR TITLE
games101-cgl: new recipe

### LINKED ISSUE #24581: [request] games101-cgl/0.1.0
### Package Name/Version

games101-cgl/0.1.0

### Webpage

https://github.com/endingly/games101-cgl

### Source code

https://github.com/endingly/games101-cgl

### Description of the library/tool

games101-cgl is a wrapper library used in homework 8 of [GAMES101](https://sites.cs.ucsb.edu/~lingqi/teaching/games101.html) which is the most popular computer graphics course in China.

This library is already available in [vcpkg](https://github.com/microsoft/vcpkg/pull/31831) since 2023/06/05 and I'm interested in expanding it to Conan.

### PR DESCRIPTION
**Pull request by FeignClaims**
### Summary

New recipe:  **games101-cgl/0.1.0**

Resolves #24581 

#### Motivation

games101-cgl is a wrapper library used in homework 8 of [GAMES101](https://sites.cs.ucsb.edu/~lingqi/teaching/games101.html) which is the most popular computer graphics course in China. However, the installation of dependencies of this library is too complicated, so it is difficult for novices to complete the homeowork correctly. By adding this recipe to conan, novices can directly use conan to complete the installation of the library.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/games101-cgl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def export_sources(self):
+        export_conandata_patches(self)
```
Comment (perseoGI) [inline recipes/games101-cgl/all/conanfile.py:32]:
```suggestion
```

There is no need to export patches when there are no patches!

File: recipes/games101-cgl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        toolchain = CMakeToolchain(self)
+        toolchain.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
```
Comment (perseoGI) [inline recipes/games101-cgl/all/conanfile.py:67]:
```suggestion
```
Same in here!

File: recipes/games101-cgl/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
+from conan import ConanFile
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir, rm
```
Comment (perseoGI) [inline recipes/games101-cgl/all/conanfile.py:2]:
```suggestion
from conan.tools.files import get, copy, rmdir, rm
```
Remove not needed methods!

**Review by perseoGI — COMMENTED** [review]
Looking good!

**Review by perseoGI — APPROVED** [review]
LGTM

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**